exporter/prometheus/prometheus.go

```diff
@@ -103,19 +103,32 @@ func (c *collector) registerViews(views ...*view.View) {
 				tagKeysToLabels(view.TagKeys()),
 				nil,
 			)
-			c.viewsMu.Lock()
-			c.newViews[sig] = desc
-			c.viewsMu.Unlock()
+			c.registeredViewsMu.Lock()
+			c.registeredViews[sig] = desc
+			c.registeredViewsMu.Unlock()
 			count++
 		}
 	}
 	if count == 0 {
 		return
 	}
 
-	if err := c.reg.Register(c); err != nil {
-		c.opts.onError(fmt.Errorf("cannot register the collector: %v", err))
-	}
+	c.ensureRegisteredOnce()
+}
+
+// ensureRegisteredOnce invokes reg.Register on the collector itself
+// exactly once to ensure that we don't get errors such as
+//    cannot register the collector: descriptor Desc{fqName: *}
```

Comment:
Bad indentation in the comments.

Reply:
That was sort of intentional, but perhaps I overdid the indentation ;)